### ISSUE TITLE
False positive in ArgumentListWrappingRule

### ISSUE THREAD

**Issue by jamespfaulkner** [pinterest/ktlint#1002]
## Expected Behavior

ktlint should not report the following violation.

## Observed Behavior
Given the following class and a `max_line_length` of `80`
```kotlin
                class B : A({
                    toInt()
                    toInt()
                    toInt()
                    toInt()
                    toInt()
                    toInt()
                })
```

ktlint fails with the following two violations:
`Argument should be on a separate line (unless all arguments can fit a single line)`
`Missing newline before ")"`

No violations are reported if there is no `max_line_length`.

## Steps to Reproduce

This can be recreated using the following test:

```kotlin
    @Test
    fun testLambdaArgumentsAreIgnored() {
        assertThat(
            ArgumentListWrappingRule().lint(
                """
                abstract class A(init: String.() -> Int)
                class B : A({
                    toInt()
                    toInt()
                    toInt()
                    toInt()
                    toInt()
                    toInt()
                })

                fun test(a: Any, b: (Any) -> Any) {
                    test(a = "1", b = {
                        it.toString()
                    })
                }
                """.trimIndent(),
                userData = mapOf("max_line_length" to "80")
            )
        ).isEmpty()
    }
```
https://github.com/pinterest/ktlint/compare/master...jamespfaulkner:false-reporting

## Your Environment
* Version of ktlint used: 0.40.0
* Version of Gradle used (if applicable): 6.6
* Operating System and version: MacOS
* Link to your project (if it's a public repository): https://github.com/pinterest/ktlint/compare/master...jamespfaulkner:false-reporting
